maven-core/src/test/java/org/apache/maven/project/PomConstructionTest.java

```diff
@@ -53,12 +57,24 @@
 
     private static String BASE_MIXIN_DIR = BASE_DIR + "/resources-mixins";
 
+    @Inject
     private DefaultProjectBuilder projectBuilder;
 
+    @Inject
     private RepositorySystem repositorySystem;
 
     private File testDirectory;
 
+    @Override
+    protected synchronized void setupContainer()
+    {
+        super.setupContainer();
+
+        ((DefaultPlexusContainer)getContainer())
```

Comment:
This requires Java 8, and would probably be clearer without lambdas

Reply:
For Maven core we already have Java 8.. 
check here version without lambda: 

https://github.com/apache/maven/blob/master/maven-embedder/src/test/java/org/apache/maven/cli/MavenCliTest.java#L325-L335